apollo/apollo.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2023 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apollo
+
+import (
+	"bytes"
+	"errors"
+	"text/template"
+
+	"github.com/apolloconfig/agollo/v4/component/log"
+	"github.com/cloudwego/kitex/pkg/klog"
+	"github.com/shima-park/agollo"
+)
+
+// Client the wrapper of apollo client.
+type Client interface {
+	SetParser(ConfigParser)
+	ClientConfigParam(cpc *ConfigParamConfig, cfs ...CustomFunction) (ConfigParam, error)
+	ServerConfigParam(cpc *ConfigParamConfig, cfs ...CustomFunction) (ConfigParam, error)
+	RegisterConfigCallback(ConfigParam, func(string, ConfigParser))
+	DeregisterConfig(ConfigParam) error
+}
+
+type ConfigParam struct {
+	Key       string
+	NameSpace string
+	Cluster   string
+	Content   string
+	Type      ConfigType
+}
+
+type client struct {
+	acli agollo.Agollo
+	// support customise parser
+	parser            ConfigParser
+	clusterTemplate   *template.Template
+	namespaceTemplate *template.Template
```

Comment:
I think there's no need of the template for namespace

Reply:
you're right